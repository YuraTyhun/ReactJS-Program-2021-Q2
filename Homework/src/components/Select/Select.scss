@import '../../assets/styles/variables.scss';

.select-sort {
    display: grid;
    grid-template-areas: "select";
    align-items: center;
    width: 100%;
    max-width: 200px;
    min-width: 170px;
    padding: 5px 7px;
    margin-left: 20px;

    select {
        appearance: none;
        background: transparent;
        color: $white;
        border: none;
        font-size: 18px;
        width: 100%;
        outline: none;
        cursor: pointer;
        padding-right: 20px;
        grid-area: select;
    }
    
    option {
        color: $dark-grey;
        cursor: pointer;
    }
    
    &::after {
        content: "";
        width: 10px;
        height: 6px;
        background-color: $pink;
        clip-path: polygon(100% 0%, 0 0%, 50% 100%);
        grid-area: select;
        justify-self: end;
    }
}