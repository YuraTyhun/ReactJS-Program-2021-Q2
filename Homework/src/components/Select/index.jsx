import React from 'react';
import PropTypes from 'prop-types';

import './Select.scss';

const Select = ({ sortOptions, onSelect }) => (
  <div className="select-sort">
    <select onChange={onSelect}>
      {sortOptions.map((option) => (
        // PATTERN: Avoid using Index as Key for map
        <option value={option.value} key={option.id}>{option.label}</option>
      ))}
    </select>
  </div>
);

Select.propTypes = {
  sortOptions: PropTypes.arrayOf(PropTypes.object),
  onSelect: PropTypes.func
};

Select.defaultProps = {
  sortOptions: [],
  onSelect: () => {}
};

export default Select;
