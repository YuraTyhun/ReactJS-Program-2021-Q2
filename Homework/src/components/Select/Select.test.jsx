import React from 'react';
import renderer from 'react-test-renderer';

import Select from './index';

it('Select renders correctly', () => {
    const options = [
        { id: 1, value: 'option 1', label: 'Test option 1' },
        { id: 2, value: 'option 2', label: 'Test option 2' },
        { id: 3, value: 'option 3', label: 'Test option 3' },
    ];
    const select = renderer
        .create(<Select sortOptions={options} onSelect={() => {}}/>)
        .toJSON();
    expect(select).toMatchSnapshot();
});