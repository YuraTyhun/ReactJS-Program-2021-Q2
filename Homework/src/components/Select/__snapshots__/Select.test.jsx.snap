// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly 1`] = `
<div
  className="select-sort"
>
  <select
    onChange={[Function]}
  >
    <option
      value="option 1"
    >
      Test option 1
    </option>
    <option
      value="option 2"
    >
      Test option 2
    </option>
    <option
      value="option 3"
    >
      Test option 3
    </option>
  </select>
</div>
`;
